source/common/logger/ur_sinks.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <ostream>
+
+namespace sink
+{
+
+class Sink
+{
+public:
+    Sink() {
+        this->fd = stdout;
+    }
+
+    Sink(std::ostream &os)
+    {
+        if (os.rdbuf() == std::cerr.rdbuf()) {
+            this->fd = stderr;
+        }
+    }
+
+    ~Sink() = default;
+
+    template<typename... Args>
+    void log(const char *level, const char *format, Args&&... args) {
```

Comment:
You could also provide a `log_concat` alternative without format, just concatenating the `std::to_string` of each argument to the log.
Useful instead of wasting time to write unneeded format strings for most use cases and avoid security holes.